contracts/RockPaperScissors.sol

```diff
@@ -10,25 +10,27 @@ contract RockPaperScissors is Pausable {
     enum Moves {ROCK, PAPER, SCISSORS}
     Moves[3] beats;
     struct Game {
-        address a;
-        address b;
-        Moves bMove;
+        address a;      // address of first player to reveal move
+        Moves aMove;    // move that was revealed
+        bool isClosed;  // true when game has two players
         uint wager;
-        uint deadline;
-        bool completed;
+        uint deadline;  // if one player has not revealed move by deadline, the other player may claim the wager
     }
     mapping(bytes32 => Game) public games;
+    mapping(bytes32 => bytes32) moves; // mapping of moves - the value is hash of game the move (key) is for
```

Comment:
This mapping contains all the moves used in games. The key is the moveHash, the value is the gameHash the move is intended for. Not sure if this is the best strategy -- input appreciated.

This structure makes it easy to prevent reuse of move hashes (just check for initialized value in mapping). However, when `startGame()` is called, the `gameHash` will be the same as the `moveHash`. It's a waste of space to store entries in `moves` where the key and value are equal (not to mention that the key in `games` is also the same). Here's the other structures I considered:

**Player 1's `moveHash` is the `gameHash`, player 2's `moveHash` is stored in the `Game` struct**
- Impossible to prevent reuse of `moveHash` because it is embedded in `Game` struct

**Only store Player 2's moveHash in `moves`**
- This saves space and is technically possible, but I found it difficult to understand my own code and it seemed highly likely to result in errors

**Use `mapping(bytes32 => bool) moves`**
- Doesn't work because player 2's move needs to be associated with a `gameHash` ahead of time to prevent cheating
- And as I understand, this actually takes a little more gas than a mapping value of `bytes32` because of bitwise shifts required

Reply:
It is good that you have this concern, a legitimate one. But if you think further, there is a way to keep the safety even when the second player submits in the clear. Can you find how.